docs/configuration/integrations/integrations-next/_index.md

```diff
@@ -0,0 +1,165 @@
+
+The revamped integrations subsystem has the following benefits over the
+original subsystem:
+
+* Integrations can opt in to supporting multiple instances. For example, you
+  may now run any number of `redis_exporter` integrations, where before you
+  could only have one per agent. Integrations such as `node_exporter` still
+  only support a single instance, as it wouldn't make sense to have multiple
+  instances of those.
+
+* Autoscrape (previously called "self-scraping"), when enabled, now supports
+  sending metrics for an integration directly to a running metrics instance.
+  This allows you configuring an integration to send to a specific Prometheus
+  remote_write endpoint.
+
+* A new service discovery HTTP API is included. This can be used with
+  Prometheus' [http_sd_config][http_sd_config]. The API returns extra labels
+  for integrations that previously were only availble when autoscraping, such
+  as `agent_hostname`.
+
+* Integrations that aren't Prometheus exporters may now be added, such as
+  integrations that generate logs or traces.
+
+[http_sd_config]: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#http_sd_config
+
+## Config changes
+
+The revamp contains a number of breaking changes to the config. The schema of the
+`integrations` key in the config file is now the following:
+
+```yaml
+integrations:
+  # Controls settings for integrations that generate metrics.
+  metrics:
+    # Controls default settings for autoscrape. Individual instances of
+    # integrations inherit the defaults and may override them.
+    autoscrape:
+      # Enables autoscrape of integrations.
+      [enable: <boolean> | default = true]
+
+      # Specifies the metrics instance name to send metrics to. Instance
+      # names are located at metrics.configs[].name from the top-level config.
+      # The instance must exist.
+      #
+      # As it is common to use the name "default" for your primary instance,
+      # we assume the same here.
+      [metrics_instance: <string> | default = "default"]
+
+      # Autoscrape interval and timeout. Defaults are inherited from the global
+      # section of the top-level metrics config.
+      [scrape_interval: <duration> | default = <metrics.global.scrape_interval>]
+      [scrape_timeout: <duration> | default = <metrics.global.scrape_timeout>]
+
+  # Override settings for agent to self-communivate for autoscrape. This is
+  # currently required if you are using TLS for the agent server. This field is
+  # temporary and will be removed in the near future once autoscrape can work #
+  # without using the network.
+  #
+  # Settings are omitted for brevity, but the schema is from:
+  # https://github.com/prometheus/common/blob/2af6d036253eee1a9a08c6ddf6be6d67537bcdff/config/http_config.go#L177
+  client_config:
+    # <settings omitted>
+
+  # Configs for integrations which do not support multiple instances.
+  [agent: <agent_config>]
```

Comment:
Hmm, the `_exporter` prefix is really just because those are the names of the projects we're embedding. It's the cadvisor integration because that's the project name, ditto with the node_exporter integration being named after node_exporter.

If we unify at all in the future, I think it's more likely we might drop the _exporter prefix instead to represent what we're integrating with rather than what project the integration originates from.

Reply:
Maybe this would be a good time to drop the `_exporter` prefix from the integrations?

Reply:
cc @rgeyer for thoughts here

Reply:
I am good with either direction, dropping the exporter feels cleaner.

Reply:
I'll leave this to a follow up PR, I think. #1231 for tracking